src/test/java/hu/elte/eserial/builder/SimpleBuilderTest.java

```diff
@@ -0,0 +1,118 @@
+package hu.elte.eserial.builder;
+
+import hu.elte.eserial.exception.EserialBuilderMismatchException;
+import hu.elte.eserial.exception.EserialPrimitiveCanNotBeNullException;
+import org.junit.Test;
+
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class SimpleBuilderTest {
+
+    @Test(expected = EserialBuilderMismatchException.class)
+    public void build_GivenInvalidType_ThrowsEserialBuilderMismatchException() {
+        new SimpleBuilder(Date.class).build(3L);
+    }
+
+    @Test(expected = EserialBuilderMismatchException.class)
+    public void build_GivenInvalidValue_ThrowsEserialBuilderMismatchException() {
+        new SimpleBuilder(Integer.class).build(new Date());
+    }
+
+    @Test(expected = EserialPrimitiveCanNotBeNullException.class)
+    public void build_GivenPrimitiveTypeNullValue_ThrowsEserialPrimitiveCanNotBeNullException() {
+        new SimpleBuilder(Integer.class).build(null);
```

Comment:
Waait, `Integer.class` should be able to accept nulls. It's `int.class` we want to pay attention to. Or am I missing something?